apps/api/src/server/applications/sector/sector.controller.js

```diff
@@ -21,12 +22,18 @@ const mapSectors = (sectors) => {
  * @type {import('express').RequestHandler}
  */
 export const getSectors = async (request, response) => {
-	const sectors = isEmpty(request.query)
-		? await sectorRepository.getAll()
-		: await subSectorRepository.getBySector({ name: request.query.sectorName?.toString() });
+	let sectors = await getCache(request.query.sectorName?.toString() || '');
 
-	// sort ascending order of subsector abbreviation, BC, EN, ... WA, WS, WW
-	sectors.sort((a, b) => (a.abbreviation > b.abbreviation ? 1 : -1));
+	if (!sectors) {
+		sectors = isEmpty(request.query)
+			? await sectorRepository.getAll()
+			: await subSectorRepository.getBySector({ name: request.query.sectorName?.toString() });
+
+		// sort ascending order of subsector abbreviation, BC, EN, ... WA, WS, WW
+		sectors.sort((a, b) => (a.abbreviation > b.abbreviation ? 1 : -1));
+
+		setCache(request.query.sectorName?.toString() || '', sectors);
```

Comment:
This can be its own whole function and is responsible for getting the sectors from the database. In theory each function should only be a few lines long

Reply:
I don't understand. This (setCache) is already a function stored in the utils folder?

Reply:
Yes, sorry I mean everything inside the `if` clause